http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1291,4 +1330,37 @@ public RequestDistribution(HttpClient httpClient, Processor processor)
             });
         }
     }
+
+    // By wrapping HttpClient, we are able to substitute the underlying AuthenticationStore
+    // with a more efficient one.
+    private static class WrappedHttpClient
+            extends HttpClient
+    {
+        private final AuthenticationStore authenticationStore;
+
+        public WrappedHttpClient(
+                ClientSecurityConfig config,
+                HttpClientTransport transport,
+                SslContextFactory sslContextFactory)
+        {
+            super(transport, sslContextFactory);
+            authenticationStore = new ExpiringAuthenticationStore(new SpnegoAuthentication(
```

Comment:
@electrum, did you mean that we did not need to create a WrappedHttpClient? Actually we do because HttpClient does not have an API to allow people to plug in a custom AuthenticationStore. We need a more efficient AuthenticationStore here than the one implemented by Jetty. Yeah, we need to create a bug for Jetty folks to fix it.

Or did you mean that we should not wrap the lines :)?